static/app/views/performance/transactionSummary/transactionReplays/useReplaysFromTransaction.tsx

```diff
@@ -18,13 +19,20 @@ export type ReplayListRecordWithTx = ReplayListRecord & {
   txEvent: {[x: string]: any};
 };
 
+import {
+  getPercentilesEventView,
+  mapPercentileValues,
+  PercentileValues,
+} from '../transactionEvents/utils';
+
 type State = Awaited<ReturnType<typeof fetchReplayList>> & {
   eventView: undefined | EventView;
   replays?: ReplayListRecordWithTx[];
 };
 
 type Options = {
   eventsWithReplaysView: EventView;
+  getFilteredEventView: (percentiles: PercentileValues) => EventView | undefined;
```

Comment:
it looks like if we changed the signature of the function so it always returns an `EventView` then we wouldn't need the other EventView prop. what do you think?

Reply:
Good call. I didn't think about that since the replay query is generating its own EventView that should be fine

Reply:
But I still need one for getPercentilesEventView. Gonna dig a little more because I did have a feeling like it could be improved somehow

Reply:
@ryan953 

I could move `getFilteredEventView` into `useReplaysFromTransaction`. 

This would require passing in 
```
  const eventsDisplayFilterName = decodeEventsDisplayFilterFromLocation(location);
  const spanOperationBreakdownFilter = decodeFilterFromLocation(location);
```

As additional args to `useReplaysFromTransaction` but we would gain more simplicity by just accepting the one eventView prop

Reply:
ah right..

maybe the call to `getPercentilesEventView` and `fetchPercentiles` should happen inside the `getFilteredEventView` fn prop.

and the Options type becomes:
```diff
  type Options = {
-   eventsWithReplaysView: EventView;
-  getFilteredEventView: (percentiles: PercentileValues) => EventView | undefined;
+  getFilteredEventView: (percentiles: PercentileValues) => EventView | Promise<EventView>;
```

Reply:
To me, moving getFilteredEventView into useReplaysFromTransaction makes more sense. If you think we should do it that way though just let me know and I'll make the commit

Reply:
@ryan953 just updated with what I had in mind let me know what you think